fmdtools/analyze/graph.py

```diff
@@ -1559,48 +1878,124 @@ class ASGGraph(Graph):
     def __init__(self, asg, withstates=True):
         self.g = nx.compose(asg.flow_graph, asg.action_graph) 
         self.set_nx_labels(asg)
-        if withstates: self.set_nx_states(asg)
+        if withstates:
+            self.set_nx_states(asg)
+
     def set_nx_labels(self, asg):
+        """
+        TODO : requires documentation
+        Parameters
+        ----------
+        asg
+
+        Returns
+        -------
+
+        """
         for n in self.g.nodes():
-            if n in asg.action_graph.nodes():   self.g.nodes[n]['label'] = 'Action'
-            elif n in asg.flow_graph.nodes():   self.g.nodes[n]['label'] = 'Flow'
+            if n in asg.action_graph.nodes():
+                self.g.nodes[n]['label'] = 'Action'
+            elif n in asg.flow_graph.nodes():
+                self.g.nodes[n]['label'] = 'Flow'
         for e in self.g.edges():
-            if e in asg.action_graph.edges():   self.g.edges[e]['label'] = 'condition'
-            elif e in asg.flow_graph.edges():   self.g.edges[e]['label'] = 'contains' 
+            if e in asg.action_graph.edges():
+                self.g.edges[e]['label'] = 'condition'
+            elif e in asg.flow_graph.edges():
+                self.g.edges[e]['label'] = 'contains'
+
     def set_nx_states(self, asg):
+        """
+        TODO : requires documentation
```

Comment:
Attaches state and fault information to the underlying graph.

asg : ASG
     Underlying action sequence graph object to get states from